pkg/policy/checker.go

```diff
@@ -0,0 +1,31 @@
+// Package policy provides a policy checker for the helmfile state.
+package policy
+
+import "fmt"
+
+// checkerFunc is a function that checks the releaseState.
+type checkerFunc func(map[string]interface{}) (isStrict bool, err error)
+
+func forbidEnvironmentsWithReleases(releaseState map[string]interface{}) (bool, error) {
+	// forbid environments and releases to be defined at the same yaml part
+	_, hasEnvironments := releaseState["environments"]
+	_, hasReleases := releaseState["releases"]
+	if hasEnvironments && hasReleases {
+		return false, fmt.Errorf("environments and releases cannot be defined at the same yaml part")
```

Comment:
```suggestion
		return false, fmt.Errorf("environments and releases cannot be defined within the same YAML part. Use --- to extract the environments into a dedicated part")
```

Nit but this might sound better!